crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -481,6 +487,34 @@ def services
           return
         end
 
+        begin
+          unless horizon_node.nil?
```

Comment:
This should not be executed if `monasca_node.nil?` also the error message in `rescue` refers to `monasca_node.name` and `horizon_node.name` which could be undefined if those nodes are not found.

Reply:
Ok, fixed. Not rebasing so you can look at the diff (back into rebase-once-all-is-tested mode :-))